include/dds/core/xtypes/detail/DynamicType.hpp

```diff
@@ -14,16 +14,66 @@
  * limitations under the License.
  *
 */
-
 #ifndef EPROSIMA_DDS_CORE_XTYPES_DETAIL_DYNAMIC_TYPE_HPP_
 #define EPROSIMA_DDS_CORE_XTYPES_DETAIL_DYNAMIC_TYPE_HPP_
 
+#include <string>
+#include <dds/core/xtypes/TypeKind.hpp>
+#include <functional>
+#include <dds/core/xtypes/Annotation.hpp>
+
 namespace dds {
 namespace core {
 namespace xtypes {
 namespace detail {
 
-class DynamicType { };
+class DynamicType
+{
+public:
+    const std::string & name() const noexcept{return name_ ;}
+    const TypeKind & kind() const noexcept{return kind_ ;}
+
+    void name(const std::string &name){ name_ = name ;}
+    void kind(const TypeKind &  kind) { kind_ = kind ;}
+
+    void annotation(
```

Comment:
Should be the detail annotation instead?

Reply:
in principle, it could be either one. I opted for this action, but it can be changed. that follows the question we asked each other in the past: do we for detail xtypes/detail files to include xtypes/ files?